sklearn/utils/estimator_checks.py

```diff
@@ -4145,3 +4145,77 @@ def check_param_validation(name, estimator_orig):
                         getattr(estimator, method)(y)
                     else:
                         getattr(estimator, method)(X, y)
+
+
+def check_set_output(name, transformer_orig):
+    # Check transformer.set_output configures the output of `transform`
+    tags = transformer_orig._get_tags()
+    if "2darray" not in tags["X_types"] or tags["no_validation"]:
+        return
+
+    rng = np.random.RandomState(0)
+    transformer = clone(transformer_orig)
+
+    X = rng.uniform(size=(20, 5))
+    X = _pairwise_estimator_convert_X(X, transformer_orig)
+    y = rng.randint(0, 2, size=20)
+    y = _enforce_estimator_tags_y(transformer_orig, y)
+    set_random_state(transformer)
+
+    X_trans_no_setting = transformer.fit_transform(X, y)
+    if name in CROSS_DECOMPOSITION:
+        X_trans_no_setting = X_trans_no_setting[0]
+
+    transformer.set_output(transform="default")
+    X_trans_default = transformer.fit_transform(X, y)
```

Comment:
Should we also test sequential `fit().transform()`?

Reply:
That makes sense. I added testing for `fit().transform()` in [`1421e8a` (#23734)](https://github.com/scikit-learn/scikit-learn/pull/23734/commits/1421e8ae232bf6f68a53361486511faaaff8f7ba)